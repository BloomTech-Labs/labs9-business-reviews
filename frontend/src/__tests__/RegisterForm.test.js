import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import RegisterFormComponent from '../components/RegisterForm';

describe('RegisterFormComponent />', () => {
  it('renders', () => {
    shallow(<RegisterFormComponent />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<RegisterFormComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain NavBar', () => {
    let app = shallow(<RegisterFormComponent/>);
    expect(app.find("NavBar").length).toBe(1);
  });
  it('should contain 3 label elements', () => {
    let app = shallow(<RegisterFormComponent/>);
    expect(app.find("label").length).toBe(3);
  });
});
