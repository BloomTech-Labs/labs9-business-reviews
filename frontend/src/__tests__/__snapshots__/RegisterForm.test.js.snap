// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterFormComponent /> matches snapshot 1`] = `
<div>
  <NavBar />
  <styled.div>
    <styled.form
      onSubmit={[Function]}
    >
      <styled.h1>
         Register for a account!
      </styled.h1>
      <label
        htmlFor="email"
      >
         Email
      </label>
      <styled.input
        name="email"
        onChange={[Function]}
        type="email"
      />
      <label
        htmlFor="password"
      >
         Password
      </label>
      <styled.input
        name="password"
        onChange={[Function]}
        type="password"
      />
      <label
        htmlFor="name"
      >
        Name
      </label>
      <styled.input
        name="name"
        onChange={[Function]}
        type="text"
      />
      <input
        className="btn"
        type="submit"
        value="Register"
      />
    </styled.form>
  </styled.div>
</div>
`;
